go/libraries/doltcore/sqle/tables.go

```diff
@@ -400,22 +407,176 @@ func (t *DoltTable) DataLength(ctx *sql.Context) (uint64, error) {
 	return numBytesPerRow * numRows, nil
 }
 
-// TODO: Have actual implementations for AnalyzeTable and Statistics; this is just a quick fix to unblock others.
-
 // AnalyzeTable implements the sql.StatisticsTable interface.
 func (t *DoltTable) AnalyzeTable(ctx *sql.Context) error {
+	table, err := t.DoltTable(ctx)
```

Comment:
now might be a good time to make either a `HistogramBuilder` or `TableStatisticsBuilder` in GMS to avoid the duplication here. A couple ways of doing that: i) builder for each table, consumes rows; ii) constructor is passed an `sql.RowIter`, it exhausts the row iter and returns `map[string]sql.TableStatistics`

Reply:
(maybe fold with info schema fix?)